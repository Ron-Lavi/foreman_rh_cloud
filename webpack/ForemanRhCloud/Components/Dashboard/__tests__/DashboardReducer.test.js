import { testReducerSnapshotWithFixtures } from '@theforeman/test';

import {
  INVENTORY_POLLING_START,
  INVENTORY_POLLING,
  INVENTORY_TAB_CHANGED,
  INVENTORY_POLLING_ERROR,
} from '../DashboardConstants';
import reducer from '../DashboardReducer';
import {
  pollingProcessID,
  logs,
  initialState,
  activeTab,
  error,
  accountID,
  scheduled,
} from '../Dashboard.fixtures';

const fixtures = {
  'should return the initial state': initialState,
  'should handle INVENTORY_POLLING_START': {
    action: {
      type: INVENTORY_POLLING_START,
      payload: {
        pollingProcessID,
        accountID,
      },
    },
  },
  'should handle INVENTORY_POLLING': {
    action: {
      type: INVENTORY_POLLING,
      payload: {
        logs,
        accountID,
        activeTab,
        scheduled,
      },
    },
  },
  'should handle INVENTORY_TAB_CHANGED': {
    action: {
      type: INVENTORY_TAB_CHANGED,
      payload: {
        activeTab,
        accountID,
      },
    },
  },
  'should handle INVENTORY_POLLING_ERROR': {
    action: {
      type: INVENTORY_POLLING_ERROR,
      payload: {
        error,
        accountID,
        activeTab,
      },
    },
  },
};

describe('Dashboard reducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
