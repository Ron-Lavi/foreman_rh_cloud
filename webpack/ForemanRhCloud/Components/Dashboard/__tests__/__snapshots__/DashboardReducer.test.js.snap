// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard reducer should handle INVENTORY_POLLING 1`] = `
Object {
  "some-account-ID": Object {
    "uploads": Object {
      "error": null,
      "logs": Array [
        "some-logs...",
      ],
      "scheduled": "2019-08-21T16:14:16.520+03:00",
    },
  },
}
`;

exports[`Dashboard reducer should handle INVENTORY_POLLING_ERROR 1`] = `
Object {
  "some-account-ID": Object {
    "uploads": Object {
      "error": "some-error",
    },
  },
}
`;

exports[`Dashboard reducer should handle INVENTORY_POLLING_START 1`] = `
Object {
  "some-account-ID": Object {
    "activeTab": "generating",
    "pollingProcessID": 1,
  },
}
`;

exports[`Dashboard reducer should handle INVENTORY_TAB_CHANGED 1`] = `
Object {
  "some-account-ID": Object {
    "activeTab": "uploads",
  },
}
`;

exports[`Dashboard reducer should return the initial state 1`] = `Object {}`;
